File: frontend/src/types/dataSource.ts
export interface DataSourceResponse {
  id: number;
  name: string;
  ownerId: number;
  dbHost: string;
  dbPort: number;
  dbName: string;
  createdAt: string;
  updatedAt: string;
}

export interface DataSourceCreateRequest {
  name: string;
  dbHost: string;
  dbPort: number;
  dbName: string;
  dbUsername: string;
  dbPassword?: string;
}

export interface DataSourceUpdateRequest {
  name?: string;
  dbHost?: string;
  dbPort?: number;
  dbName?: string;
  dbUsername?: string;
  dbPassword?: string;
}

export interface SchemaRefreshSummary {
  tableCount: number;
  columnCount: number;
  cachedAt: string;
}

File: backend/src/main/java/com/querygenie/entity/DataSource.java
package com.querygenie.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.Instant;

@Entity
@Table(name = "data_sources")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DataSource {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @Column(name = "db_host", nullable = false, length = 255)
    private String dbHost;

    @Column(name = "db_port", nullable = false)
    private Integer dbPort;

    @Column(name = "db_name", nullable = false, length = 100)
    private String dbName;

    @Column(name = "encrypted_credentials", nullable = false, columnDefinition = "TEXT")
    private String encryptedCredentials;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "schema_cache", columnDefinition = "jsonb")
    private String schemaCache;

    @Column(name = "schema_cached_at")
    private Instant schemaCachedAt;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;
}
